OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -934,7 +934,144 @@ internal static void UpgradeActors(int engineVersion, ref List<MiniYamlNode> nod
 		{
 			foreach (var node in nodes)
 			{
-				// Add rules here
+				// Fix RA building footprints to not use _ when it's not necessary
+				if (engineVersion < 20160530 && Game.ModData.Manifest.Mod.Id == "ra")
+				{
+					// Tesla Coil
+					if (node.Value.Value == "tsla")
+					{
+						var locationNode = node.Value.Nodes.First(n => n.Key == "Location");
```

Comment:
I think the cleanest solution would be to define a `Dictionary<string, CVec> locationOffsets` for each mod and then do the manipulation if it `ContainsKey(node.Value.Value)`.